gen/generate.go

```diff
@@ -79,28 +113,43 @@ func GenerateGenesisFiles(genesisData GenesisData, consensus config.ConsensusPro
 	}
 
 	if sum != TotalMoney {
-		panic(fmt.Sprintf("Amounts don't add up to TotalMoney - off by %v", int64(TotalMoney)-int64(sum)))
-	}
-
-	// Backwards compatibility with older genesis files: if the consensus
-	// protocol version is not specified, default to V0.
-	proto := genesisData.ConsensusProtocol
-	if proto == protocol.ConsensusVersion("") {
-		proto = protocol.ConsensusCurrentVersion
+		fsum := float64(sum)
+		ftot := float64(TotalMoney)
+		if (math.Abs((fsum-ftot)/ftot) < 0.01) && (u64absDiff(sum, TotalMoney) < 10000) {
+			if verbose {
+				fmt.Fprintf(verboseOutWriter, "doing roundoff fixup expected total money %d actual sum %d\n", TotalMoney, sum)
+			}
+			// wallet stake is a float and roundoff might happen but we might be close enough to do fixup
```

Comment:
do you know why wallet stake is a float while genesis stake is an int64? shouldn't wallet stake also be represented in int64 to avoid floating-point problem?

Reply:
It's a float64 percent; [0.0 .. 100.0]; for being human-convenent.